src/installation/index.md

```diff
@@ -1,61 +1,167 @@
 [espressif/llvm-project]: https://github.com/espressif/llvm-project
 [GCC toolchain]: https://github.com/espressif/crosstool-NG/
 [tracking issue]: https://github.com/espressif/llvm-project/issues/4
 [channel]: https://rust-lang.github.io/rustup/concepts/channels.html
 
-This chapter, will cover how to properly install the correct Rust compiler and toolchain for our ESP chips.
+### espup
+
+[esp-rs/espup] is a tool for installing and maintaining the necessary ecosystem to develop applications in Rust for Espressif SoC's (both `Xtensa` and `RISC-V` targets).
+
+`espup` takes care of installing the proper Rust compiler (our fork in case of `Xtensa` targets and the `nightly` toolchain with the necessary target for `RISC-V` targets), `LLVM` toolchain,  `GCC` toolchains, `rustup`, and many other things. For more details, [see Usage section of the `espup` Readme].
+
+To install `espup`, use the following command:
+```sh
+cargo install espup
+```
+You can also directly download pre-compiled [release binaries] or use [cargo-binstall].
+
+Once that `espup` is installed, you can simply run:
+```sh
+espup install
+```
+
+This will install all the necessary tools to develop Rust applications for all supported ESP targets.
+
+`espup` will create an export file, by default `$HOME/export-esp.sh` on Unix systems and `%USERPROFILE%\export-esp.ps1` on Windows. This file contains some environment variables required to build projects. Make sure to source this file in every terminal before building any application:
+
+```sh
+# Unix
+. $HOME/export-esp.sh
+# Windows
+%USERPROFILE%\export-esp.ps1
+```
+
+[esp-rs/espup]: https://github.com/esp-rs/espup
+[see Usage section of the `espup` Readme]: https://github.com/esp-rs/espup#usage
+[release binaries]: https://github.com/esp-rs/espup/releases
+[cargo-binstall]: https://github.com/cargo-bins/cargo-binstall
+
+### Other installation methods
+
+- Using [esp-rs/rust-build] installation scripts. This was the recommended way in the past, but now the installation scripts are feature frozen and all new features will only be included in `espup`. See the repository README for instructions.
+- Building the Rust compiler with `Xtensa` support from source. This process is computationally expensive and can take one or more hours to complete depending on your system. It is not recommended unless there is a major reason to go for this approach. See instructions in the [Installing from Source section of the esp-rs/rust repository].
+
+[esp-rs/rust-build]: https://github.com/esp-rs/rust-build
+[Installing from Source section of the esp-rs/rust repository]: https://github.com/esp-rs/rust#installing-from-source
+
+## Rust with `std` runtime
+
+Regardless of the target architecture, if you want to build a project using the [`std` approach], you will also need:
+- [ESP-IDF]: Espressif IoT Development Framework as it's used as our hosted environment.
+  - See [ESP-IDF installation note] for details on how ESP-IDF can be installed.
```

Comment:
Maybe this can be expanded on, you don't actually need to clone  ESP-IDF at all do you? I think that `embuild` clones it into the `.embuild` directory?